### DISCUSSION TITLE
Doppio controllo PWM Teensy 4.1+ T41U5XBB

### DISCUSSION THREAD

**Discussion by distebia** [grblHAL/iMXRT1062#75] [Q&A]
Salve, ho visto che con la build del 10/12/23 è stata abilitata la possibilità di avere + mandrini (anche pwm) collegati e gestiti da grblhal. Il teensy e la T41U5XBB sono in grado di gestire 2 uscite pwm indipendenti? se si quali pin potrei usare?

**Comment by terjeio** [comment]
[PWM capable aux outputs](https://github.com/grblHAL/iMXRT1062/blob/b2daa21ab3ef4e54a4d80ff36dad8cc3011e2b4e/grblHAL_Teensy4/src/ioports_analog.c#L49-L129) can be used by redefining one as AUXOUTPUT0_PWM_PIN in the map and enable a spindle driver on top of that with SPINDLE_PWM2. Note that such a spindle can not be used for laser mode and that I have not tested this.
Alternatively a plugin can be written that implement a fully capable PWM spindle, again by using available aux outputs.
